packages/api-i18n/src/graphql/graphql/locales.ts

```diff
@@ -142,15 +150,20 @@ const plugin: GraphQLSchemaPlugin<I18NContext & SecurityContext> = {
                         return new NotFoundResponse(`Locale "${args.code}" not found.`);
                     }
 
-                    await i18n.locales.update(code, {
-                        default: args.default
-                    });
+                    if (locale.default && !data.default) {
+                        return new ErrorResponse({
+                            message:
+                                "Cannot unset default locale, please set another locale as default first."
+                        });
+                    }
 
-                    if (locale.default) {
+                    await i18n.locales.update(code, data);
+
+                    if (data.default) {
                         await i18n.locales.updateDefault(code);
                     }
-
-                    return new Response(locale);
+                    const updatedLocale = await i18n.locales.getByCode(code);
```

Comment:
If we'd migrate the logic into crud file, we'd probably be able to avoid this extra DB call.